internal/pkg/piecemanager/storage_miner_back_end.go

```diff
@@ -0,0 +1,124 @@
+package piecemanager
+
+import (
+	"context"
+	"io"
+
+	"github.com/filecoin-project/go-storage-miner"
+
+	"github.com/pkg/errors"
+)
+
+// StorageMinerAPI provides an interface to the bits of a storage miner required
+// by the StorageMinerBackEnd, which satisfies PieceManager interface.
+type StorageMinerAPI interface {
+	// AllocateSectorID allocates a new sector ID.
+	AllocateSectorID() (sectorID uint64, err error)
+
+	// PledgeSector allocates a new sector, fills it with self-deal junk, and
+	// seals that sector.
+	PledgeSector() error
+
+	// SealPiece writes the provided piece to a newly-created sector which it
+	// immediately seals.
+	SealPiece(ctx context.Context, size uint64, r io.Reader, sectorID uint64, dealID uint64) error
+
+	// GetSectorInfo produces information about a sector managed by this storage
+	// miner, or an error if the miner does not manage a sector with the
+	// provided identity.
+	GetSectorInfo(sectorID uint64) (storage.SectorInfo, error)
+
+	// ListSectors lists all the sectors managed by this storage miner (sealed
+	// or otherwise).
+	ListSectors() ([]storage.SectorInfo, error)
+}
+
+// SectorBuilderAPI defines the subset of the sector builder API required by the
+// StorageMinerBackEnd.
+type SectorBuilderAPI interface {
```

Comment:
Needing this behavior from the sector builder directly feels off.  The production implementation of the storage miner already holds onto this so we should be able to expose these things in the right way.

Reply:
@ZenGround0 - In this context, we differ in what "feels off" and the "right way" to do things; I will need more guidance if I am to make your software dreams come true. Would you please be more specific with what you'd like to see changed?

Reply:
we hath resolved this offline (online?)